### ISSUE TITLE
validation loss not decreasing and predictions have offsets

### ISSUE THREAD

**Issue by AnishZz** [matterport/Mask_RCNN#2613]
Hello all, First of all thank you @Waleedka for the maskrcnn implementation that was provided, and maintained very well
throughout these years.

My issue stated here, is regarding overfitting,finetuning and prediction offsets on the custom dataset.

The dataset used is real-estate memorandum images and only one target object which is 'valid'.
'valid' in our use-case was the useful text portion in the real-estate memorandum image. you can get an idea by below shown
three images.

![3d2eba101005008b7d5ddb7736440db9-p_h](https://user-images.githubusercontent.com/43930955/123538689-bbec1200-d753-11eb-8c9b-cd5a3087a7cf.jpg)
![el-cerrito-development-projects-offering-memorandum-1-638](https://user-images.githubusercontent.com/43930955/123538691-bdb5d580-d753-11eb-91d9-6c800f5f1c6b.jpg)
![page_1](https://user-images.githubusercontent.com/43930955/123538694-bee70280-d753-11eb-98ea-86c48b387e93.jpg)

Dataset size- 1000 images for training, 22 images for validation.
Annotations used- viatool rectangular region annotations.

Number of Epochs trained- 100
Layers- Only Heads
training loss- 0.37
validation loss- 1.41
learning rate - 0.0002

So, the resulting inferences on the validation set are shown below

![3bc6783f-970d-4b0c-a955-57953f8a1b80](https://user-images.githubusercontent.com/43930955/123538750-ec33b080-d753-11eb-8107-b00c7f684d79.png)
![6685a69d-b368-4eeb-944e-60fb3d874e9a](https://user-images.githubusercontent.com/43930955/123538755-edfd7400-d753-11eb-801e-f38978cb3e6a.png)
![ae487a82-8950-49c5-8125-17a83a088173](https://user-images.githubusercontent.com/43930955/123538756-efc73780-d753-11eb-86c4-4eb65f2f52b8.png)

you can observe the masks are not properly occupying the bounding boxes and the bounding boxes are some times offset to
some portion of the predictions.
Also after 100th Epoch, the training loss and validation loss are not at all decreasing. i checked upto 40 Epochs.
Increasing the learning rate lead to huge increase in training loss and validation loss
Decreasing the learning rate lead to no improvement in training loss,validation loss and predictions.

It would be great if anyone who faced the similar issue one way or the other could kindly
share their suggestions and references. 

Best Regards,

AnishZz

**Comment by alexander-pv** [comment]
Hi, you can try to change a threshold in `unmold_mask` function in `./mrcnn/utils.py` 
You can also use bounding boxes instead of masks from the output.
Perhaps, you can get rid of color channels in your task.
In general, text detection tasks can be solved by more classical cv-algorithms with OpenCV. 
You can also check text detection nn-models like [this](https://github.com/argman/EAST).

**Comment by AnishZz** [comment]
Hi @alexander-pv . Thanks for your reply.
I will try out the changing a threshold in unmold_mask function in ./mrcnn/utils.py, and let you know.
I have also used only bboxes without masks from output, but the offset problem is same as previous.
Yes, i've tried EAST model, there we would get only individual letters and words get detected. But here i want sentences, chunks of words to be detected in a certain way as per the problem statement.

Thanks for your suggestions @alexander-pv
